core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -65,10 +65,13 @@ private[ui] class AllJobsPage(parent: JobsTab, store: AppStatusStore) extends We
     }.map { job =>
       val jobId = job.jobId
       val status = job.status
-      val jobDescription = store.lastStageAttempt(job.stageIds.max).description
-      val displayJobDescription = jobDescription
-        .map(UIUtils.makeDescription(_, "", plainText = true).text)
-        .getOrElse("")
+      var displayJobDescription = ""
+      try {
+        displayJobDescription = store.lastStageAttempt(job.stageIds.max).description
+          .map(UIUtils.makeDescription(_, "", plainText = true).text).getOrElse("")
+      } catch {
+        case e => displayJobDescription = job.description.getOrElse("")
```

Comment:
No, you're catching all exceptions. This should check whether the last stage attempt exists rather than catching anything that goes wrong.

I don't think it's correct to pretend it exists but is empty. The request is for something that doesn't exist and ideally generates a 404.